src/snapred/backend/recipe/ReductionRecipe.py

```diff
@@ -0,0 +1,158 @@
+from typing import Any, Dict, List, Tuple, Type
+
+from snapred.backend.dao.ingredients import ReductionIngredients as Ingredients
+from snapred.backend.log.logger import snapredLogger
+from snapred.backend.recipe.ApplyNormalizationRecipe import ApplyNormalizationRecipe
+from snapred.backend.recipe.GenerateFocussedVanadiumRecipe import GenerateFocussedVanadiumRecipe
+from snapred.backend.recipe.PreprocessReductionRecipe import PreprocessReductionRecipe
+from snapred.backend.recipe.Recipe import Recipe, WorkspaceName
+from snapred.backend.recipe.ReductionGroupProcessingRecipe import ReductionGroupProcessingRecipe
+from snapred.meta.Config import Config
+from snapred.meta.decorators.Singleton import Singleton
+
+logger = snapredLogger.getLogger(__name__)
+
+Pallet = Tuple[Ingredients, Dict[str, str]]
+
+
+@Singleton
+class ReductionRecipe(Recipe[Ingredients]):
+    """
+    Currently requires:
+        Ingredients:
+            self.maskList = ingredients.maskList
+            self.pixelGroup = ingredients.pixelGroup
+            self.smoothingParameter = ingredients.smoothingParameter
+            self.detectorPeaks = ingredients.detectorPeaks
+
+        Groceries:
+            self.sampleWs = groceries["inputWorkspace"]
+            self.diffcalWs = groceries.get("diffcalWorkspace", "")
+            self.normalizationWs = groceries.get("normalizationWorkspace", "")
+            ~~self.backgroundWs = groceries.get("backgroundWorkspace", "")~~
+            self.groupingWS = groceries["groupingWorkspace"]
+    """
+
+    PRESERVE_INTERMEDIATE_WORKSPACES = Config["cis_mode"]
```

Comment:
The `MakeDirtyDish` algorithm and the `WashDishes` algorithm should be callable from this recipe